src/components/questionPreviews/SelectOneQuestionPreview/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { useMemo } from 'react';
+import {
+    _cs,
+    isNotDefined,
+    noOp,
+} from '@togglecorp/fujs';
+import { gql, useQuery } from '@apollo/client';
+import {
+    RadioInput,
+    TextOutput,
+} from '@the-deep/deep-ui';
+
+import {
+    SingleOptionListQuery,
+    SingleOptionListQueryVariables,
+} from '#generated/types';
+
+import styles from './index.module.css';
+
+const SINGLE_OPTION_LIST = gql`
+    query SingleOptionList(
+        $projectId: ID!,
+        $choiceCollectionId: ID!,
+        ) {
+        private {
+            projectScope(pk: $projectId) {
+                choiceCollection(pk: $choiceCollectionId) {
+                    label
+                    id
+                    name
+                    choices {
+                        id
+                        label
+                        name
+                    }
+                    id
+                }
+            }
+        }
+    }
+`;
+
+const choiceCollectionKeySelector = (d: { id: string }) => d.id;
+const choiceCollectionLabelSelector = (d: { name: string }) => d.name;
```

Comment:
Let's give proper type to this.

Reply:
it gives error when given proper type @AdityaKhatri